dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1639,6 +1643,52 @@ public synchronized void flush() throws IOException {
         }
     }
 
+    public static class ThreadLocalByteArrayOutputStream extends OutputStream {
+        private final static int BYTE_ARRAY_OUTPUT_BUFFER_SIZE = 128;
+        private final ThreadLocal<ByteArrayOutputStream> threadLocal = new ThreadLocal<ByteArrayOutputStream>() {
+            @Override
+            protected ByteArrayOutputStream initialValue() {
+                return new ByteArrayOutputStream(BYTE_ARRAY_OUTPUT_BUFFER_SIZE);
+            }
+        };
+
+        @Override
+        public void write(int b) throws IOException {
+            threadLocal.get().write(b);
+
+        }
+
+        public void reset() {
+            if (threadLocal.get().size() > BYTE_ARRAY_OUTPUT_BUFFER_SIZE)
+                threadLocal.remove();
+            threadLocal.get().reset();
```

Comment:
should only need to do the reset in cases where you didn't just discard the BAOS.
```
            if (threadLocal.get().size() &gt; BYTE_ARRAY_OUTPUT_BUFFER_SIZE) {
                threadLocal.remove();
            } else {
                threadLocal.get().reset();
            }
```